### ISSUE TITLE
no-wait-for-side-effects does not catch returned side effect

### ISSUE THREAD

**Issue by zaicevas** [testing-library/eslint-plugin-testing-library#351]
This yields an error:

```
    await waitFor(() => {
      userEvent.click(screen.getByText('Continue'))
    })
```

This does not:

```
    await waitFor(() => userEvent.click(screen.getByText('Continue')))
```

Is that expected behaviour?

**Comment by Belco90** [comment]
Hey @zaicevas! It's not an expected behavior indeed. It seems the plugin is only looking for arrow functions with a body but ignoring those with an implicit return.
